port/common/omrportcontrol.c

```diff
@@ -54,26 +54,37 @@ omrport_control(struct OMRPortLibrary *portLibrary, const char *key, uintptr_t v
 		return 0;
 	}
 #if defined(OMR_ENV_DATA64)
-	if (!strcmp(OMRPORT_CTLDATA_ALLOCATE32_COMMIT_SIZE, key)) {
+	if (0 == strcmp(OMRPORT_CTLDATA_ALLOCATE32_COMMIT_SIZE, key)) {
 		if (0 != value) {
-			/* CommitSize is immutable. It can only be set once. */
```

Comment:
The immutability check has been removed. Now, no error will be thrown if the value is set more than once. Will memory allocation functions work correctly if these values are changed more than once?

Reply:
For `INCREMENT_SIZE`, changing the value only affects the size of the newly reserved region that the sub-allocator will add. This won't cause any issues with iterating the heap to find a block later on.

Looking at the code that uses `COMMIT_SIZE`, I couldn't reason as to why the immutability check would be needed here. We only ever call `omrport_control` with these options once when creating the heap. And there is nothing stopping us otherwise from just setting the `PPG` value directly. Moreover, the commit size within the code can be changed depending on the values of other variables.